### PR TITLE
Support Cloning From Private Git Repository

### PR DESCRIPTION
**Pull request by martinmaly**
* API accepts secret ref for clone from unregistered repository
* add options to test git server
* test server implements basic auth

### CODE REVIEW

**Review by justinsb — APPROVED** [review]
A few random thoughts, but nothing blocking.  lgtm!

File: porch/repository/pkg/git/testing.go [hunk 1]
```diff
@@ -98,6 +108,14 @@ func (s *GitServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 // serveRequest is the main dispatcher for http requests.
 func (s *GitServer) serveRequest(w http.ResponseWriter, r *http.Request) error {
+	if s.username != "" || s.password != "" {
+		username, password, ok := r.BasicAuth()
+		if !ok || username != s.username || password != s.password {
```
[the remark this answers is not in the document]
Comment (martinmaly) [reply porch/repository/pkg/git/testing.go:113]: thanks!

File: porch/engine/pkg/engine/clone_test.go [hunk 2]
```diff
@@ -143,13 +144,31 @@ func startGitServer(t *testing.T, repo *gogit.Repository) string {
 	return fmt.Sprintf("http://%s", address)
 }
 
-type cr struct{}
+type cr struct {
```
[the remark this answers is not in the document]
Comment (martinmaly) [reply porch/engine/pkg/engine/clone_test.go:147]: :) done.

File: porch/engine/pkg/engine/clone.go [hunk 3]
```diff
@@ -80,8 +85,7 @@ func (m *clonePackageMutation) cloneFromGit(ctx context.Context, gitPackage *api
 	}
 	defer os.RemoveAll(dir)
 
-	// TODO: Add support for authentication.
-	var credentialResolver repository.CredentialResolver = nil
+	var credentialResolver repository.CredentialResolver = m.credentialResolver
```
[the remark this answers is not in the document]
Comment (martinmaly) [reply porch/engine/pkg/engine/clone.go:88]: thanks! fixed.

**Pull request merged**